File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.stream.*;
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        int n = numbers.length;
        String[] stringNumbers = new String[n];
        
        for(int i = 0; i < n; i++) {
            stringNumbers[i] = String.valueOf(numbers[i]);
        }
        
        /*
         * String[] stringArray = Arrays.stream(numbers) // IntStream 생성
         *                     .mapToObj(String::valueOf) // int 값을 String으로 변환
         *                     .toArray(String[]::new); // Stream을 String 배열로 수집
         */
        
        Arrays.sort(stringNumbers, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));
        
        // System.out.println(Arrays.toString(stringNumbers));
        
        return getAnswer(stringNumbers);
    }
    
    private String getAnswer(String[] numbers) {
        if(numbers[0].equals("0")) {
            return "0";
        }
        
        return String.join("", numbers);
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    answer = ''
    
    ## 첫번째 풀이 : 조합으로 최댓값 찾기 했지만 런타임에러로 제출 실패 ㅜ
    # from itertools import permutations
    # nums = list(permutations(numbers, len(numbers)))
    # num_list = list(''.join(map(str,n))  for n in nums)

    
    # 두번째 풀이 : (블로그 참고)
    # [3,30,34] 처럼 숫자의 길이가 달라 정렬 시키면 [34,30,3] -> 34303 으로 나와서 땡
    # 각 숫자마다 3번씩 반복 시킴 [343434,303030,333] 여기서 앞 세자리로 비교하는 것으로 구현
    # numbers.sort(key = lambda x : (str(x)*3), reverse = True)
    # for n in numbers :
    #       answer += str(n)
    

    numbers = list(map(str, numbers))               # string으로 바꾸어서 비교
    numbers.sort(key = lambda x : x*3,reverse=True)  # 3번씩 반복하면 붙였을때 큰수 찾기 가능
    
    for i in numbers:             # 정렬된 리스트를 answer에 순서대로 더해줌
        answer += i

    return str(int(answer))